src/core_plugins/kibana/public/dashboard/directives/dashboard_panel.js

```diff
@@ -69,52 +57,89 @@ uiModules
+       * @type {function}
+       */
+      saveState: '&'
     },
     link: function ($scope, element) {
       if (!$scope.panel.id || !$scope.panel.type) return;
 
-      loadPanel($scope.panel, $scope)
-      .then(function (panelConfig) {
-        // These could be done in loadPanel, putting them here to make them more explicit
-        $scope.savedObj = panelConfig.savedObj;
-        $scope.editUrl = panelConfig.editUrl;
+      /**
+       * Initializes the panel for the saved object.
+       * @param {{savedObj: SavedObject, editUrl: String}} savedObjectInfo
+       */
+      function initializePanel(savedObjectInfo) {
+        $scope.savedObj = savedObjectInfo.savedObj;
+        $scope.editUrl = savedObjectInfo.editUrl;
 
         element.on('$destroy', function () {
-          panelConfig.savedObj.destroy();
-          $scope.parentUiState.removeChild(getPersistedStateId(panelConfig.panel));
+          $scope.savedObj.destroy();
           $scope.$destroy();
         });
 
         // create child ui state from the savedObj
-        const uiState = panelConfig.uiState || {};
-        $scope.uiState = $scope.parentUiState.createChild(getPersistedStateId(panelConfig.panel), uiState, true);
-        const panelSavedVis = _.get(panelConfig, 'savedObj.vis');  // Sometimes this will be a search, and undef
-        if (panelSavedVis) {
-          panelSavedVis.setUiState($scope.uiState);
+        const uiState = $scope.savedObj.uiStateJSON ? JSON.parse($scope.savedObj.uiStateJSON) : {};
+        $scope.uiState = $scope.createChildUiState({ path : getPersistedStateId($scope.panel), uiState });
+
+        if ($scope.savedObj.vis) {
+          $scope.savedObj.vis.setUiState($scope.uiState);
+          $scope.savedObj.vis.listeners.click = $scope.getVisClickHandler();
+          $scope.savedObj.vis.listeners.brush = $scope.getVisBrushHandler();
         }
 
         $scope.filter = function (field, value, operator) {
           const index = $scope.savedObj.searchSource.get('index').id;
           filterManager.add(field, value, operator, index);
         };
-      })
-      .catch(function (e) {
-        $scope.error = e.message;
 
-        // If the savedObjectType matches the panel type, this means the object itself has been deleted,
-        // so we shouldn't even have an edit link. If they don't match, it means something else is wrong
-        // with the object (but the object still exists), so we link to the object editor instead.
-        const objectItselfDeleted = e.savedObjectType === $scope.panel.type;
-        if (objectItselfDeleted) return;
+        // This causes changes to a saved search to be hidden, but also allows
+        // the user to locally modify and save changes to a saved search only in a dashboard.
+        // See https://github.com/elastic/kibana/issues/9523 for more details.
+        $scope.panel.columns = $scope.panel.columns || $scope.savedObj.columns;
```

Comment:
This used to only happen for saved searches. Are $`scope.panel.columns` and `$scope.panel.sort` only being set so that we can watch them and know when to call `saveState`?